### ISSUE TITLE
Error in lgb.check.obj(params, obj) when using quantile regression.

### ISSUE THREAD

**Issue by AraneoA** [Microsoft/LightGBM#1281]
Hi,
I am trying to do a quantile regression and get an error stating that the objective is invalid. I understand that quantile regresison is available in the current version installed v2.1.0

Since the data I´m working with is too big I am pasting a small reproducible example below plus additional details.

Thanks!

## Environment info
Operating System: UBUNTU 17.10
CPU: AMD Ryzen/CUDA GPU
C++/Python/R version: R version 3.4.2 (2017-09-28)

## Error Message:
lgb.check.obj: objective name error should be one of (regression, regression_l1, regression_l2, huber, fair, poisson, binary, lambdarank, multiclass)

## Reproducible examples
library(lightgbm)

#Generate data
set.seed(123)
N  <- 10000
X1 <- rnorm(N, 0, 3)
X2 <- rnorm(N, 50, 15)
Y <- 25*sin(X1) + 0.3*X2 + 150 + rnorm(N, 0, 10)
data <- data.frame(X1, X2, Y)

#Split test/train
train_ind <- sample(seq_len(nrow(data)), size = 7000)
train.x <- as.matrix(data[train_ind, c("X1", "X2") ])
train.y <- as.matrix(data[train_ind, "Y" ])
test.x <- as.matrix(data[-train_ind, c("X1", "X2") ])
test.y <- as.matrix(data[-train_ind, "Y" ])

#Fit data
dtrain <- lgb.Dataset(train.x, label = train.y)
model <- lgb.train(data = dtrain,
                   objective = "quantile",
                   alpha = 0.1,
                   rounds = 1000,
                   min_data = 1,
                   learning_rate = .1)

#Display prediction, real value, error
View(cbind(predict(model, test.x), test.y, predict(model, test.x) - test.y) )

#Calculate mean of errors (should be skewed if quanile regression is working)
mean(predict(model, test.x) - test.y)

**Comment by AraneoA** [comment]
Thanks! After building with latest code it is working. Must have been a recent change.
